client/plugins/utils.js

```diff
@@ -136,6 +136,47 @@ Vue.prototype.$parseCronExpression = (expression) => {
   }
 }
 
+Vue.prototype.$getNextScheduledDate = (expression) => {
+  const cronFields = expression.split(' ');
+  const currentDate = new Date();
+  const nextDate = new Date();
+
+  // Calculate next minute
+  const minute = cronFields[0];
+  const currentMinute = currentDate.getMinutes();
+  const nextMinute = getNextValue(minute, currentMinute, 59);
+  nextDate.setMinutes(nextMinute);
+
+  // Calculate next hour
+  const hour = cronFields[1];
+  const currentHour = currentDate.getHours();
+  const nextHour = getNextValue(hour, currentHour, 23);
+  nextDate.setHours(nextHour);
+
+  // Calculate next day of month
+  const dayOfMonth = cronFields[2];
+  const currentDayOfMonth = currentDate.getDate();
+  const lastDayOfMonth = new Date(currentDate.getFullYear(), currentDate.getMonth() + 1, 0).getDate();
+  const nextDayOfMonth = getNextValue(dayOfMonth, currentDayOfMonth, lastDayOfMonth);
+  nextDate.setDate(nextDayOfMonth);
+
+  // Calculate next month
+  const month = cronFields[3];
+  const currentMonth = currentDate.getMonth() + 1;
+  const nextMonth = getNextValue(month, currentMonth, 12);
+  nextDate.setMonth(nextMonth - 1);
+
+  // Calculate next day of week
+  const dayOfWeek = cronFields[4];
+  const currentDayOfWeek = currentDate.getDay();
+  const nextDayOfWeek = getNextValue(dayOfWeek, currentDayOfWeek, 6);
+  const daysUntilNextDayOfWeek = getNextDaysUntilDayOfWeek(currentDate, nextDayOfWeek);
+  nextDate.setDate(currentDate.getDate() + daysUntilNextDayOfWeek);
```

Comment:
This causes issues with `0 12 1 * *`, which is running on the 1st of every month.

The correct result is Match 1st but we are getting March 28th.